repository: two minor improvements

sourcejedi opened the pull request:
Two very minor improvements, for your reviewing pleasure.
- Clean up fds of segments we delete (during compaction)
- io.write_commit() already implies io.close_segment()

sourcejedi committed:
Clean up fds of segments we delete (during compaction)

When we delete a segment, let's close its fd as well.
Note as well wasting the fd, this was forcing the
filesystem to preserve the deleted file until we exited.

I noticed roughly 20 open fds of deleted files when
attic saved 10G of data.

sourcejedi committed:
io.write_commit() already implies io.close_segment()

ThomasWaldmann commented:
Nice findings!

One slight style improvement: use "is not None" rather than "!= None".
As there is only one None object, you can compare based on object identity rather than object value.

ThomasWaldmann commented:
also, while you are at it:

```
    self.fds.pop(segment).close()
```

Search that line and see it should use same code as you did and rather check the return value of the pop operation before calling .close() on it. Maybe never a problem in practice, but the check should be made always, imho.

sourcejedi committed:
Style fix in added code

PEP8 says to prefer "is not None"

sourcejedi committed:
recover_segment(): don't assume we have an fd for segment

Suggested by @ThomasWaldmann.  Avoiding a complex assumption
should make the code easier to understand and maintain.

(Technically we do have an fd for the segment, because
the only caller opens the segment and checks it before
calling for repair.)

sourcejedi commented:
Thanks for the style hint!
